orderer/consensus/smartbft/chain_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright IBM Corp. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package smartbft_test
+
+import (
+	"bytes"
+	"encoding/binary"
+	"testing"
+	"time"
+
+	cb "github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric-protos-go/msp"
+	"github.com/hyperledger/fabric/protoutil"
+	. "github.com/onsi/gomega"
+)
+
+var nonce uint64 = 0
+
+func TestBftChain(t *testing.T) {
+	g := NewWithT(t)
+	channelId := "channel-test"
+	const eventuallyTimeout = 10 * time.Second
+
+	t.Run("starting 4 nodes, submitting a tx, restarting all the nodes and submitting another tx should add both txs to the ledger", func(t *testing.T) {
```

Comment:
can we please have them in separate functions? No need to put a sub-function, this is not Ginkgo.